packages/x-date-pickers/src/tests/testValidation/testTextFieldValidation.tsx

```diff
@@ -213,7 +213,7 @@ function testTextFieldValidation(ElementToTest, propsToTest, getOptions) {
         expect(screen.getByRole('textbox')).to.have.attribute('aria-invalid', 'false');
       });
     }
-    if (propsToTest.includes('maxTime')) {
+    if (propsToTest.includes('maxTime') && (getOptions().isLegacyPicker || getOptions().withTime)) {
       it('should apply maxTime', function test() {
         const { render, withTime } = getOptions();
 
```

Comment:
Maybe easier to read
```suggestion
    if (propsToTest.includes('maxTime')) {
      it('should apply maxTime', function test() {
        const { render, withTime, isLegacyPicker } = getOptions();
        if(!isLegacyPicker && !withTime){
        	return 
        }

```